qiskit_optimization/applications/max_k_cut.py

```diff
@@ -0,0 +1,247 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""An application class for the Max-k-cut."""
+
+from typing import List, Dict, Tuple, Optional, Union
+import networkx as nx
+import numpy as np
+from docplex.mp.model import Model
+
+from qiskit.utils import algorithm_globals
+from qiskit_optimization.algorithms import OptimizationResult
+from qiskit_optimization.problems.quadratic_program import QuadraticProgram
+from qiskit_optimization.translators import from_docplex_mp
+from .graph_optimization_application import GraphOptimizationApplication
+
+try:
+    from matplotlib.pyplot import cm
+    from matplotlib.colors import to_rgba
+
+    _HAS_MATPLOTLIB = True
+except ImportError:
+
+    _HAS_MATPLOTLIB = False
+
+
+class Maxkcut(GraphOptimizationApplication):
+    """Optimization application for the "max-k-cut" [1] problem based on a NetworkX graph.
+
+    References:
+        [1]: Z. Tabi et al.,
+             "Quantum Optimization for the Graph Coloring Problem with Space-Efficient Embedding"
```

Comment:
This paper addresses a graph coloring problem and does not mention max-k-cut problem.
Could you refer an appropriate paper or page?

Reply:
Hi @t-imamichi, graph colouring problem is a decision variant of max-k-cut. Graph coloring problem is about checking if there "exists colouring of the nodes which make every edge connecting nodes with a different colour." Contrary, Max-K-Cut is about finding the colouring for which the number of such edges is maximized. You could say that graph colouring to max-K-Cut shows similar relation as SAT to MAX-SAT. For this reason, and the fact max cut is already used in the package we decided to choose max-k-cut name

Reply:
As far as the paper mentions only coloring problem, I don't think it's an appropriate citation. Why not rename the class "k-coloring problem" as the paper addresses?

Reply:
Moreover, there are many papers directly addressing max-k-cut problem. Why do you need to refer that paper (Tabi et al.) for max-k-cut class?

Reply:
The notion of feasibility changes. For Max-Cut every solution is feasible, for graph colouring only those with all edges connecting different nodes are feasible. This is obviously correct but inconsistent with max-cut used in the qiskit-optimization. 
Since this is for optimization purposes mostly we found Max-K-Cut more accurate. But we can also move it to Max-K-Cut
We are not aware of older papers that propose the given formulation. If there is one, we're open to replacing the reference.

Reply:
You may perhaps try to use an solution of coloring problem as a heuristic solution of max-k-cut problem?
If so, I ask you to make a proper mixed integer programming model of max-k-cut problem.
The model of application classes should be exact. Otherwise, even if we apply CPLEX or Gurobi to the model, we may not be able to obtain the optimal solution.